src/librustc_driver/lib.rs

```diff
@@ -698,13 +698,32 @@ pub trait CompilerCalls<'a> {
 
     // Create a CompilController struct for controlling the behaviour of
     // compilation.
-    fn build_controller(&mut self, _: &Session, _: &getopts::Matches) -> CompileController<'a>;
+    fn build_controller(self: Box<Self>, _: &Session, _: &getopts::Matches) -> CompileController<'a>;
 }
 
-// CompilerCalls instance for a regular rustc build.
+/// CompilerCalls instance for a regular rustc build.
 #[derive(Copy, Clone)]
 pub struct RustcDefaultCalls;
 
+pub enum AdHocCalls<'a> {
```

Comment:
Could you document `AdHocCalls` please? It's not exactly clear to me what it does - is it purely a convenience to avoid implementing `build_controller`?

Reply:
Yes